tools/validator.py

```diff
@@ -52,6 +52,86 @@ def _CheckRegistryKeyPath(self, filename, artifact_definition, key_path):
 
     return result
 
+  def _CheckWindowsPath(self, filename, artifact_definition, source, path):
+    """Checks if a path is a valid Windows path.
+
+    Args:
+      filename (str): name of the artifacts definition file.
+      artifact_definition (ArtifactDefinition): artifact definition.
+      source (SourceType): source definition.
+      path (str): path to validate.
+
+    Returns:
+      bool: True if the Windows path is valid.
+    """
+    result = True
+
+    number_of_forward_slashes = path.count('/')
+    number_of_backslashes = path.count('\\')
+    if (number_of_forward_slashes < number_of_backslashes and
+        source.separator != '\\'):
+      logging.warning((
+          'Incorrect path separator: {0:s} in path: {1:s} defined '
+          'by artifact definition: {2:s} in file: {3:s}').format(
+              source.separator, path, artifact_definition.name,
+              filename))
+      result = False
+
+    if source.separator == '\\':
```

Comment:
I'd go
if source.separator != '\\':
  \# Can't do any further validations
  return True

and move all the remaining code one level out.